src/schema/v2/Match.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { GraphQLFieldConfig } from "graphql"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { pageable } from "relay-cursor-paging"
+import { ResolverContext } from "types/graphql"
+import { ArticleType } from "./article"
+import { ArtistType } from "./artist"
+import { ArtworkType } from "./artwork"
+import { CollectionType } from "./collection"
+import { FairType } from "./fair"
+import { FeatureType } from "./Feature"
+import {
+  connectionWithCursorInfo,
+  paginationResolver,
+} from "./fields/pagination"
+import { GeneType } from "./gene"
+import { pageType } from "./page"
+import { ProfileType } from "./profile"
+import { SaleType } from "./sale"
+import { SearchMode } from "./search"
+import { SearchEntity } from "./search/SearchEntity"
+import { ShowType } from "./show"
+import { TagType } from "./tag"
+
+const MODELS = {
+  Article: { loader: "articleLoader", type: ArticleType },
+  Artist: { loader: "artistLoader", type: ArtistType },
+  Artwork: { loader: "artworkLoader", type: ArtworkType },
+  Collection: { loader: "collectionLoader", type: CollectionType },
+  Fair: { loader: "fairLoader", type: FairType },
+  Feature: { loader: "featureLoader", type: FeatureType },
+  Gene: { loader: "geneLoader", type: GeneType },
+  Page: { loader: "pageLoader", type: pageType },
+  Profile: { loader: "profileLoader", type: ProfileType },
+  Sale: { loader: "saleLoader", type: SaleType },
+  Show: { loader: "showLoader", type: ShowType },
+  Tag: { loader: "tagLoader", type: TagType },
+}
+
+export const MatchType = new GraphQLUnionType({
+  name: "Match",
+  types: Object.values(MODELS).map(({ type }) => type),
+  resolveType: ({ __typename }) => {
+    return MODELS[__typename].type
+  },
+})
+
+const matchArgs = {
+  term: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  entities: {
+    type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(SearchEntity))),
+    description: "ARTIST_SERIES, CITY, and VIEWING_ROOM are not yet supported",
+  },
+  mode: {
+    type: SearchMode,
+    description: "Mode of search to execute",
+    defaultValue: "SITE",
+  },
+  size: { type: GraphQLInt, defaultValue: 10 },
+  page: { type: GraphQLInt, defaultValue: 1 },
+}
+
+export const Match: GraphQLFieldConfig<void, ResolverContext> = {
+  type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(MatchType))),
+  args: matchArgs,
+  resolve: async (_root, args, { searchLoader, ...loaders }) => {
+    const { body } = await searchLoader(args)
+
+    return body.map(async ({ id, label }) => {
+      const loader = loaders[MODELS[label].loader]
+      const body = await loader(id)
```

Comment:
This loads each item one at a time, can there be a `Promise.all` instead?

Reply:
I think await inside of the map and Promise.all on the outside are equivalent. I'll confirm though cuz it does look odd. You can't promise all cuz I need to augment the returned value with the type.

Reply:
Oh right, if you're mapping over an async function it's gonna return a promise regardless. Funny though this does exactly what I want as is; I think because it's returning an array of promises, GraphQL is doing the correct thing.